### PR TITLE
Updating LOIO visualization

### PR DESCRIPTION
**Pull request by gwaybio**
I include the following adjustments:

1. Refactoring LOIO_evaluation.ipynb to summarize the new LOIO analysis (updated since #46 )
2. Adding new feature space performance visualizations (adding to panel A, changing structure of panel B)
3. Focusing only on CP features in panels C and D

Note, this figure will also likely update as we continue writing and compiling the story. I consider the LOIO analysis wrapped up at this point, however :tada:

### CODE REVIEW

**Comment by gwaybio** [comment]
I'm taking both Jenna and Cameron to review - please do so when you have an opportunity and the brain mood. **There is no rush!!**

(I should clarify that we only need one review, no need for both!)

File: 3.evaluate_model/scripts/nbconverted/LOIO_evaluation.py [hunk 1]
```diff
@@ -92,21 +92,40 @@ def compute_avg_rank_and_pvalue(grouped_df):
 
 
 # Calculate average rank for each Metadata_DNA
```
Comment (jenna-tomkinson) [inline 3.evaluate_model/scripts/nbconverted/LOIO_evaluation.py]: Is this comment still relevant to the below group? It looks like "Metadata_DNA" is apart of the group so if everything below also apart of "Metadata_DNA"?

Recommend clarifying this comment.

File: 7.figures/figures/main_figure_4_loio.png
Comment (jenna-tomkinson) [inline 7.figures/figures/main_figure_4_loio.png:1]: Some comments I have:

1. For panel C, there seems to be a weird difference in the bar size between correct prediction TRUE and FALSE. For example, in the `Shuffled:FALSE` section, the "MetaphaseAlignment" phenotypic class has 8 correct predictions and 17 incorrect predictions, but the correct prediction has a larger blue line than the incorrect prediction. I would think that the line would be proportional, but this could be due to a misunderstanding on my part. I appreciate any clarification 😄 
2. In panel D, it looks like the x-axis is overlapping on top of each other (e.g., 1.0 on the left over the 0.0 on the right). I would recommend avoiding this overlap just for aesthetic reasons.

**Review by jenna-tomkinson — APPROVED** [review]
LGTM! I have a few minor comments about the figure to address but overall a very interesting visualization of the LOIO assessment 😄

File: 7.figures/figures/main_figure_4_loio.png
Comment (gwaybio) [reply 7.figures/figures/main_figure_4_loio.png:1]: Sorry for the extreme delay, but I am finally now ready to address this!

> For panel C, there seems to be a weird difference in the bar size between correct prediction TRUE and FALSE. For example, in the Shuffled:FALSE section, the "MetaphaseAlignment" phenotypic class has 8 correct predictions and 17 incorrect predictions, but the correct prediction has a larger blue line than the incorrect prediction. I would think that the line would be proportional, but this could be due to a misunderstanding on my part. I appreciate any clarification 😄

Thank you so much for the detailed look into the figure! :detective: The `Correct prediction: TRUE` facet contains a different set of samples than the `Correct prediction: FALSE` facet. Specifically `TRUE` are samples that are all correct predictions for a given phenotype and `FALSE` are all incorrect predictions for that same phenotype. Therefore, we would expect 15x more samples in the incorrect predictions group (every sample has a prediction per phenotype). This plot is showing of all predictions, correct and incorrect, what proportion passes a strict threshold?

> In panel D, it looks like the x-axis is overlapping on top of each other (e.g., 1.0 on the left over the 0.0 on the right). I would recommend avoiding this overlap just for aesthetic reasons.

Great catch! I will address in next commit :)

File: 3.evaluate_model/scripts/nbconverted/LOIO_evaluation.py (hunk 1 above)
Comment (gwaybio) [reply 3.evaluate_model/scripts/nbconverted/LOIO_evaluation.py]: Great catch! The cell prior contains a deeper description, so I will delete this comment in the next commit. (I don't know if there was a markdown cell prior!)

**Comment by gwaybio** [comment]
Thanks again for the review @jenna-tomkinson - Given the extensive updates to this PR, would you be able to give this another look? Please focus on the figure, and I hope that it will not take too much time :)

main_figure_4_loio

>  I consider the LOIO analysis wrapped up at this point, however 🎉

:point_up: Haha, silly me :upside_down_face: 

-------
Edit: In the most recent commit, I removed panel A, to be moved to the supplement.

File: 7.figures/figures/main_figure_4_loio.png
Comment (jenna-tomkinson) [inline 7.figures/figures/main_figure_4_loio.png:1]: This figure looks so much better and tells the story in a more clean and concise manner!

I only have one question:

Are these figures showing the result from the balanced or unbalanced model, with or without IC, and is this with all nuclei features (not a subset for AreaShape)?

This will probably be stated in the figure legend, but wanted to ask since I am curious what model I am looking at.

**Review by jenna-tomkinson — APPROVED** [review]
Love the changes to this figure! I left one question on the figure for you to address, but overall this PR is great and ready to merge!

File: 7.figures/figures/main_figure_4_loio.png
Comment (gwaybio) [reply 7.figures/figures/main_figure_4_loio.png:1]: Great questions! This is our standard model: Balanced, with IC, and all features. I add clarification in 8618141f06f5183f2c47c73e33e4806bad0770ca

Going to merge now, thanks again!
